### ISSUE TITLE
Support for gitiles web urls different from repo url

### ISSUE THREAD

**Issue by ganli** [reduckted/GitWebLinks#311]
Our self-hosted gitiles web url is something like: https://google.co/plugins/gitiles/policy/alice
HTTPS clone url is: https://google.co/a/policy/alice
SSH clone url is: ssh://google.co:29418/policy/alice

Previously we clone with ssh and configure gitWebLinks as:
```
		{
			"http": "https://google.co/plugins/gitiles",
			"ssh": "ssh://google.co:29418"
		},
```

But now we are forced to use HTTPS to clone repo. As our HTTPS clone url is different from gitiles HTTP url, is there any way to configure gitWebLinks to support this?

Thanks in advance!

**Comment by reduckted** [comment]
@ganli I've made a change to the VS Code extension in #312 that allows you to specify an extra setting to define the URL of the web interface for Gitiles. Would you be able to try it out and confirm that it solves the problem for you? You can download the extension here:

https://github.com/reduckted/GitWebLinks/actions/runs/10870371274/artifacts/1934534794

The extra setting is the `web` property. For your example, you would want to use these settings:
```
{
    "http": "https://google.co/a",
    "ssh": "ssh://google.co:29418",
    "web": "https://google.co/plugins/gitiles"
},
```

Once you've confirmed it works, I'll finish off the PR by making the same change to the Visual Studio extension.

**Comment by ganli** [comment]
Thanks @reduckted 
I tried the patched extension you fixed, it works smoothly.
Really great work! Then I could continue using this extension to share code with team.😁

**Comment by reduckted** [comment]
This is now available in v2.12.0 from the Visual Studio marketplace.
